src/3d/symbols/qgspointcloud3dsymbol.cpp

```diff
@@ -367,3 +367,147 @@ void QgsRgbPointCloud3DSymbol::setBlueContrastEnhancement( QgsContrastEnhancemen
+  {
+    QDomElement catElem = doc.createElement( QStringLiteral( "category" ) );
+    catElem.setAttribute( QStringLiteral( "value" ), QString::number( category.value() ) );
+    catElem.setAttribute( QStringLiteral( "label" ), category.label() );
+    catElem.setAttribute( QStringLiteral( "color" ), QgsSymbolLayerUtils::encodeColor( category.color() ) );
+    catElem.setAttribute( QStringLiteral( "render" ), category.renderState() ? "true" : "false" );
+    catsElem.appendChild( catElem );
+  }
+  elem.appendChild( catsElem );
+}
+
+void QgsClassificationPointCloud3DSymbol::readXml( const QDomElement &elem, const QgsReadWriteContext &context )
+{
+  Q_UNUSED( context )
+
+  mPointSize = elem.attribute( "point-size", QStringLiteral( "2.0" ) ).toFloat();
+  mRenderingParameter = elem.attribute( "rendering-parameter", QString() );
+
+  const QDomElement catsElem = elem.firstChildElement( QStringLiteral( "categories" ) );
+  if ( !catsElem.isNull() )
+  {
+    mCategoriesList.clear();
+    QDomElement catElem = catsElem.firstChildElement();
+    while ( !catElem.isNull() )
+    {
+      if ( catElem.tagName() == QLatin1String( "category" ) )
+      {
+        const int value = catElem.attribute( QStringLiteral( "value" ) ).toInt();
+        const QString label = catElem.attribute( QStringLiteral( "label" ) );
+        const bool render = catElem.attribute( QStringLiteral( "render" ) ) != QLatin1String( "false" );
+        const QColor color = QgsSymbolLayerUtils::decodeColor( catElem.attribute( QStringLiteral( "color" ) ) );
+        mCategoriesList.append( QgsPointCloudCategory( value, color, label, render ) );
+      }
+      catElem = catElem.nextSiblingElement();
+    }
+  }
+}
+
+QString QgsClassificationPointCloud3DSymbol::renderingParameter() const
+{
+  return mRenderingParameter;
+}
+
+void QgsClassificationPointCloud3DSymbol::setRenderingParameter( const QString &parameter )
+{
+  mRenderingParameter = parameter;
+}
+
+void QgsClassificationPointCloud3DSymbol::setCategoriesList( const QgsPointCloudCategoryList &categories )
+{
+  mCategoriesList = categories;
+}
+
+QgsPointCloudCategoryList QgsClassificationPointCloud3DSymbol::getFilteredOutCategories() const
+{
+  QgsPointCloudCategoryList filteredOut;
+  for ( QgsPointCloudCategory category : mCategoriesList )
+  {
+    if ( !category.renderState() )
+      filteredOut.push_back( category );
+  }
+  return filteredOut;
+}
+
+QgsColorRampShader QgsClassificationPointCloud3DSymbol::colorRampShader() const
+{
+  QgsColorRampShader colorRampShader;
+  colorRampShader.setColorRampType( QgsColorRampShader::Type::Exact );
+  colorRampShader.setClassificationMode( QgsColorRampShader::ClassificationMode::Continuous );
+  QList<QgsColorRampShader::ColorRampItem> colorRampItemList;
+  for ( const QgsPointCloudCategory &category : mCategoriesList )
+  {
+    QColor color = category.color();
+    color.setAlphaF( category.renderState() ? 1.0f : 0.0f );
```

Comment:
Can we drop this now and just skip those unwanted categories?

Reply:
Yes, we don't need that line anymore